api4/oauth.go

```diff
@@ -538,6 +538,7 @@ func completeOAuth(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	w.Header().Set("Content-Type", "text/html; charset=utf-8")
```

Comment:
`http.Redirect` is supposed to do this automatically, unless `Content-Type` is already set. But I can't find anywhere in this handler where we set it? Are we sure this is the endpoint that needs to be changed?

Can we add a unit test to verify all of this?

Reply:
It's set by [ServeHTTP](https://github.com/mattermost/mattermost-server/blob/master/web/handlers.go#L94). I think we might need something to prevent that call though since it seems like we'll be setting that header twice

Reply:
When you set it two times it get overwritten so I think is not a huge deal.

about the testing, is quite complicated because the current tests aren't testing the thing properly (In my opinion) because it needs some kind of external service to work, and we are not mocking that. So... I would this is clearly not straight forward to test it. and I don't want to make a huge refactor to fix this bug. We can add a User Story to rethink the OAuth testing, but I think is out of the scope of this bug.

Reply:
Now that I think of it, I'm pretty sure this is how the file download endpoints work anyway since they also go through that  write their own content type same branch of ServeHTTP